torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -227,9 +278,21 @@ def creation_fn(self, all_args, *, is_runtime: bool):
         return rebuilt
 
     def make_runtime_safe(self):
+        def _make_size_runtime_safe(x: Union[None, int, torch.SymInt]) -> Optional[int]:
+            dummy = -1
```

Comment:
This will leak into `NJT (since `__tensor_unflatten__()` will now see `-1` in the `outer_size` list, but @soulitzer has carefully written `NJT.__tensor_unflatten__()` to ignore it). I know we talked about it and I think this is reasonable - but it would be good to have something like:
```
Note [NJT outer_size in AOTDispatcher]
```

here, so you can add a link to that note in the impl, making it clear to people reading the NJT code what the expected behavior is: https://github.com/pytorch/pytorch/blob/main/torch/nested/_internal/nested_tensor.py#L267C57-L267C67

(in particular, the invariant is that unflatten should **only** see `-1` for the jagged_dim outer_size when the input tensors in question are real tensors, not FakeTensors)

Reply:
@guilhermeleobas bump, can you add a comment somewhere in `NestedTensor.__torch_dispatch__` that refers back to the code here?

Reply:
Oh I see that you added the comment below - I also think it would be helpful if you include something like:
```
See Note: [NJT outer_size in AOTDispatcher]
```

directly in `NestedTensor.__torch_dispatch__`, so if someone is debugging code there and wondering why there is a `-1` they can find the paper trail

Reply:
Forgot about this one! Adding now.